search-app/src/cmr/search/services/community_usage_metrics/metrics_service.clj

```diff
@@ -33,51 +34,40 @@
   (when (>= col 0)
     (nth csv-line col)))
 
-(defn- get-short-name-by-entry-title
-  "Query elastic for a collection with a given entry-title, returns short-name"
-  [context entry-title]
-  (when (seq entry-title)
-    (let [condition (qm/string-condition :entry-title (str entry-title "*") false true)
-          query (qm/query {:concept-type :collection
-                           :condition condition
-                           :page-size 1
-                           :all-revisions? true
-                           :result-format :query-specified
-                           :result-fields [:short-name]})
-          results (qe/execute-query context query)
-          short-name (first (map :short-name (:items results)))]
-      short-name)))
-
-(defn- get-collection-by-short-name
-  "Query elastic for collection with a given short-name, returns short-name"
-  [context short-name]
-  (when (seq short-name)
-    (let [condition (qm/string-condition :short-name short-name)
-          query (qm/query {:concept-type :collection
-                           :condition condition
-                           :page-size 1
-                           :all-revisions? true
-                           :result-format :query-specified
-                           :result-fields [:short-name]})
-          results (qe/execute-query context query)
-          short-name (first (map :short-name (:items results)))]
-      short-name)))
+(defn- get-collection-by-product-id
+  "Query elastic for a collection with a given product-id, parses out the value before the last : and
+   checks that value against both entry-title or short-name. Also checks the non-parsed value against
+   short-name."
+  [context product-id]
+  (when (seq product-id)
+    (when-let [parsed-product-id (as-> (re-find #"(^.+):|(^.+)" product-id) matches
+                                       (remove nil? matches)
+                                       (last matches))]
+      (let [condition (gc/or (qm/string-condition :entry-title parsed-product-id false false)
+                             (qm/string-condition :short-name parsed-product-id false false)
+                             (qm/string-condition :short-name product-id false false))
+            query (qm/query {:concept-type :collection
+                             :condition condition
+                             :page-size 1
+                             :all-revisions? true
+                             :result-format :query-specified
+                             :result-fields [:short-name]})
+            results (qe/execute-query context query)
+            short-name (first (map :short-name (:items results)))]
```

Comment:
every item found is guaranteed to have a `short-name`,  so you could just use `(:short-name (first (:items results)))` and put this on line 57.